features/permissions/permissions.step-definitions.js

```diff
@@ -0,0 +1,152 @@
+const assert = require('assert');
+const { Given, When, Then } = require('cucumber');
+const ENV = require('../../environment-variables');
+const request = require('request-promise-native');
+const btoa = require('btoa');
+let accessToken = "a"; // a variable to store the access token
+
+Given('I have a valid credentials, for permissions', function () {
+    this.username = ENV.USERNAME;
+    this.password = ENV.PASSWORD;
+});
+
+When('Attempt to get an access token, for permissions', async function () {
+    const clientCredentials = `${this.username}:${this.password}`;
+    try {
+        this.responseJson = await request({
+            method: 'POST',
+            uri: ENV.OAUTH_URL,
+            form: {
+                client_id: 'client-super',
+                client_secret: 'secret',
+                grant_type: 'client_credentials'
+            },
+            headers: {
+                Authorization: `Basic ${btoa(clientCredentials)}`
+            }
+        });
+        accessToken = JSON.parse(this.responseJson)["access_token"];
+    } catch (error) {
+        this.responseJson = error.error;
+    }
+});
+
+When('I attempt to get permissions list from the API', async function () {
+    try {
+        this.responseJson = await request({
+            method: 'GET',
+            uri: ENV.PERMISSIONS_LIST_URL,
+            headers: {
+                Authorization: `Bearer ${accessToken}`
+            }
+        });
+    } catch(error) {
+        this.responseJson = error.error;
+    }
+});
+
+When('I attempt to get a single permission with id 1 from the API', async function () {
+    try {
+        this.responseJson = await request({
+            method: 'GET',
+            uri: ENV.GET_PERMISSION_URL,
+            headers: {
+                Authorization: `Bearer ${accessToken}`
+            }
+        });
+    } catch(error) {
+        this.responseJson = error.error;
+    }
+});
+
+When('I make a POST request to permissions', async function () {
+    try {
+        this.responseJson = await request({
```

Comment:
Instead of having to pass in all of this stuff to request, it would be nice if we had some helper functions like `request.get()`, `request.post()` etc that would take some of the boilerplate out of the tests themselves.